src/datasets/low_carbon_london/preprocess_lcl.py

```diff
@@ -0,0 +1,244 @@
+import csv
+import logging
+import os
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+
+def get_current_month_end(df: pd.DataFrame, date_col="dt"):
+    """
+    Get the month end of current month.
+    Note: Pandas offsets.MonthEnd()
+    returns the date of the following month instead of the current month.
+
+    E.g. if dt = 31st Jan, offsets.MonthEnd returns 28th Feb
+    instead of 31st Jan. To work around this, we offset by -1 days
+    to get previous day, then do offsets.MonthEnd()
+
+    This is faster than offsets.MonthEnd().rollforward(ts):
+    - This method doesn't work on array, only works on raw Timestamp object,
+    which will require for loops (or .apply) which is super slow on big data
+    - More efficient to perform this in a vectorised manner by offset
+    with -1 days then use offsets.MonthEnd().
+
+    Args:
+        df (Pandas dataframe): Input pandas dataframe
+        date_col (str, optional): Name of date column. Defaults to "dt".
+
+    Returns:
+        pd.DataFrame: Output dataframe with the "month_end" column
+    """
+    df["month_end"] = df[date_col] + pd.offsets.Day(-1)
+    df["month_end"] = df["month_end"] + pd.offsets.MonthEnd()
+    df["month_end"] = df["month_end"].dt.date
+    return df
+
+
+def load_data(file_path: str) -> pd.DataFrame:
+    """
+    Load LCL data from csv
+
+    Args:
+        files_path (str): Folder containing CSV files
+
+    Returns:
+        pd.DataFrame: LCL dataset
+    """
+    logger.info(f"🚛 Loading data from {file_path}")
+    df = pd.read_csv(file_path)
+    df = df.rename(columns={"KWH/hh (per half hour) ": "kwh"})
+    return df
+
+
+def extract_date_features(
+    df: pd.DataFrame,
+) -> pd.DataFrame:
+    """
+    Parse date features from DateTime column.
+
+    Args:
+        df (pd.DataFrame): Input dataframe
+
+    Returns:
+        pd.DataFrame: Output dataframe with date features
+    """
+    logger.info("📅 Extracting date features")
+    df["dt"] = pd.to_datetime(df["DateTime"])
+    df["date"] = df["dt"].dt.date.astype(str)
+    df["month"] = df["dt"].dt.month.astype(int)
+    df["week"] = df["dt"].dt.to_period("W-MON").dt.start_time
+    df = get_current_month_end(df, date_col="dt")
+    df["month_max"] = pd.to_datetime(df["month_end"]).dt.day.astype(int)
+    df["day"] = df["dt"].dt.day.astype(int)
+    df["dayofweek"] = df["dt"].dt.dayofweek.astype(int)
+    df["hour"] = df["dt"].dt.hour.astype(int)
+    df["minute"] = df["dt"].dt.minute.astype(int)
+    return df
+
+
+def parse_settlement_period(
```

Comment:
❓ does this take into account the daylight savings changes?  do we care?

Reply:
LCT dataset are all given in UTC, so we don't need to worry about it. I'll add a comment in the code.

Reply:
Updated comments and readme.md clarifying that LCL dataset are all given in UTC.